.github/workflows/go-tests.yml

```
@@ -13,24 +15,123 @@ on:
+        shell: bash
+        run: |
+          echo $GITHUB_CONTEXT
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+
       - name: Check out code
         uses: actions/checkout@v4
-      - name: Run tests
-        uses: ./go/actions/test
         with:
-          run-code-checks: true
+          # For PRs, this ensures that we do not end up in a detached HEAD state;
+          # For pushes, this does not change anything.
+          # We require this to be able to easily push changes, if needed.
+          ref: ${{ github.event.pull_request.head.ref || '' }}
+          # We need HEAD^ to determine the list of changed files
+          fetch-depth: 2
+
+      # Determine whether only .ql files changed: if so, we won't run the macOS and Windows workflows
+      - name: Determine whether only .ql files changed
+        id: only-ql-check
+        shell: bash
+        run: |
+          CHANGES=$(git diff --name-only HEAD^)
+          echo $CHANGES
+
+          ONLY_QL_CHANGES=true
+          for change in $(git diff --name-only HEAD^)
+          do
+            if [[ $change != go/ql/* ]];
+            then
+              ONLY_QL_CHANGES=false
+              echo "Files other than .ql files have changed"
+              break
+            fi
+          done
+
+          echo "Setting ONLY_QL_CHANGES to $ONLY_QL_CHANGES"
+          echo "ONLY_QL_CHANGES=$ONLY_QL_CHANGES" >> $GITHUB_OUTPUT
+
+      # Dependabot will nuke Bazel build files when it updates Go dependencies;
+      # this will restore them
+      - name: Regenerate Bazel files for Dependabot PR
+        if: github.event.pull_request.user.login == 'dependabot[bot]'
+        shell: bash
+        run: |
+          bazel run //go:gazelle
+          bazel run //go:gen
+
+      # And this will push the Bazel files to the PR branch
+      - name: Push Bazel files for Dependabot PR
+        id: push-bazel-files
+        if: github.event.pull_request.user.login == 'dependabot[bot]'
+        shell: bash
+        run: |
+          git add go/extractor/vendor/**
+
+          if git diff --exit-code HEAD;
+          then
+            echo "Regenerate Bazel files resulted in no changes"
+          else
+            echo "Pushing regenerated Bazel files"
+            git config user.name "github-actions[bot]"
+            git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+            git commit -m 'Regenerate Bazel build files for Dependabot PR'
+            git push
+          fi
+
+          echo "TEST_REF=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+
+  run-linux-tests:
+    name: "Go: Run Linux tests"
+    runs-on: ubuntu-latest
+    needs:
+      - prepare-go-tests
+    steps:
+      - name: Dispatch test workflow
+        shell: bash
+        run: |
+          gh workflow run "go-tests-linux.yml" --repo "${{ github.repository }}" --ref "${{ github.head_ref || github.ref_name }}" --json
+        env:
+          GH_TOKEN: ${{ github.token }}
```

doesn't a workflow call work here?
```suggestion
    uses: ./.github/go-tests-linux.yml
    with:
      ref: ${{ needs.prepare-go-tests.outputs.ref }}
```
Otherwise I'm not sure the dispatched workflow will be very visible from the PR